internal/backend/workloadproxy/healthcheck.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+package workloadproxy
+
+import (
+	"context"
+	"net"
+	"slices"
+	"syscall"
+	"time"
+
+	"github.com/siderolabs/go-loadbalancer/upstream"
+	"go.uber.org/zap"
+)
+
+type remote struct {
+	Logger   *zap.Logger
+	Addr     string
+	AddrPort string
+}
+
+func (r *remote) Equal(other *remote) bool { return r.AddrPort == other.AddrPort }
+
+func (r *remote) HealthCheck(ctx context.Context) (upstream.Tier, error) {
+	start := time.Now()
+	err := r.healthCheck(ctx)
+	elapsed := time.Since(start)
+
+	return calcTier(err, elapsed)
+}
+
+func (r *remote) healthCheck(ctx context.Context) error {
+	d := probeDialer()
+
+	c, err := d.DialContext(ctx, "tcp", r.AddrPort)
+	if err != nil {
+		r.Logger.Warn("healthcheck failed", zap.String("address", r.AddrPort), zap.Error(err))
```

Comment:
I'm not sure we should log that. Might be too verbose.

Reply:
Moved down to Debug level.